### ISSUE TITLE
The shape of trainx error

### ISSUE THREAD

**Issue by jiansheng-li** [LennartPurucker/finetune_tabpfn_v2#24]
When we use DataParallel, we find that the shape of train_xs may be incorrect. The DataParallel seem to split n_samples from one dataset into 8gpus, however we expect to split 8 datasets into 8gpus. Because the test_x need to use the whole train_x as the in-context samples.

**Comment by LennartPurucker** [comment]
I am not sure I understand your problem correctly. Looping in @iivalchev 

This codebase is set up to work on a single downstream task/dataset. Thus, one dataset is split repeatedly (each element of a batch).

Given this notation, are you saying that the problem is that DP parallelizes one split of one dataset across the GPUs? Instead of parallelizing eight splits of one dataset across the GPUs?

**Comment by jiansheng-li** [comment]
Thanks a lot. I’m trying to fine-tune multiple datasets simultaneously.

**Comment by LennartPurucker** [comment]
This code base does not support this out of the box. You would need to adapt the data loader such that each batch consists of multiple datasets.

**Comment by iivalchev** [comment]
If not mistaken `DataPrallel` parallelism is at batch level which is challenging when you try to use multiple datasets. And yes the current implementation does not support multiple datasets. Supporting this is not that trivial and depends your data. 

Also the current strategy for splitting the single data set and using it for training is not obvious. Which I suspect is causing the confusion. The dataset is split into folds 10 by default. So 9 folds are used as training set and 1 is used as test set, folds are rotated so that each fold is used as test set. Depending on the bach size those splits are placed in batches. If `DataParallel` is used you get parallel fine-tuning across those splits (items in the batch are run in parallel).

Hope that helps.

**Comment by jiansheng-li** [comment]
Thanks. I misunderstood the task. I'm trying to fine-tune multiple datasets.

**Comment by LennartPurucker** [comment]
Closing this issue, feel free to re-open for more discussions.
